sdk/python/feast/field.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from feast.protos.feast.core.Feature_pb2 import FeatureSpecV2 as FieldProto
+from feast.types import FeastType, from_value_type
+
+
+class Field:
+    """
+    A Field represents a set of values with the same structure.
+
+    Attributes:
+        name: The name of the field.
+        dtype: The type of the field, such as string or float.
+    """
+
+    name: str
+    dtype: FeastType
+
+    def __init__(
+        self, *, name: str, dtype: FeastType,
+    ):
+        """
+        Creates a Field object.
+
+        Args:
+            name: The name of the field.
+            dtype: The type of the field, such as string or float.
+        """
+        self.name = name
+        self.dtype = dtype
+
+    def __eq__(self, other):
+        if self.name != other.name or self.dtype != other.dtype:
+            return False
+        return True
```

Comment:
Do you need to implement `__hash__` here as well? Usually that and `__eq__` need to be implemented together.